File: selection_sort/python/selection_sort.py
#!/usr/bin/env python3

from typing import List

def selection_sort(arr: List, size: int) -> None:
  for i in range(size):
    j, k = i, i

    while j < size:
      if arr[j] > arr[k]:
        k = j
      j += 1

    arr[i], arr[k] = arr[k], arr[i]
  return None

if __name__ == "__main__":
  arr = [-13, 1, 6, -3, 0, 5, -2, 10]
  selection_sort(arr, len(arr))

File: selection_sort/java/SelectionSort.java

public class SelectionSort {

  public static int[] SelectionSort(int[] array) {

    for (int i = 0; i < array.length; i++) {
      int k = i;

      for (int j = i + 1; j < array.length; j++) {
        if (array[j] > array[k]) {
          k = j;
        }
      }

      int t = array[i];
      array[i] = array[k];
      array[k] = t;
    }

    return array;
  }
}
